### PR TITLE
ci: upgrade notes

### PR DESCRIPTION
**Pull request by xkostka2**
* added check.yml workflow
* added .commitlintrc.json with specific rules
* changed README.md for consistency

### CODE REVIEW

File: .releaserc.json [hunk 1]
```diff
@@ -1,21 +1,58 @@
...
-    { "name": "alpha", "prerelease": true },
-    { "name": "beta", "prerelease": true }
-  ],
+  "branches": ["production", "+([0-9])?(.{+([0-9]),x}).x"],
   "preset": "conventionalcommits",
   "plugins": [
     "@semantic-release/commit-analyzer",
-    "@semantic-release/release-notes-generator",
-    ["@semantic-release/npm", {
-        "npmPublish": false
```
Comment (melanger) [inline .releaserc.json:13]: Keep npm, it increments version number in package.json

File: .releaserc.json [hunk 2]
```diff
@@ -1,21 +1,58 @@
...
+      "@semantic-release/changelog",
+      {
+        "changelogFile": "UPGRADE.md",
+        "changelogTitle": "Upgrade notes"
+      }
+    ],
+    [
+      "@semantic-release/git",
+      {
+        "assets": ["UPGRADE.md"],
```
Comment (melanger) [inline .releaserc.json:53]: Keep "dist"

File: .releaserc.json [hunk 3]
```diff
@@ -1,21 +1,58 @@
...
     "@semantic-release/commit-analyzer",
-    "@semantic-release/release-notes-generator",
-    ["@semantic-release/npm", {
-        "npmPublish": false
-    }],
-    ["@semantic-release/git", {
-      "assets": ["dist"],
-      "message": "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}"
-    }],
-    "@semantic-release/github"
```
Comment (melanger) [inline .releaserc.json:19]: Keep, it creates GitHub release

File: .releaserc.json [hunk 4]
```diff
@@ -1,21 +1,58 @@
 {
-  "branches": [
-    "production",
-    "+([0-9])?(.{+([0-9]),x}).x",
-    { "name": "alpha", "prerelease": true },
```
Comment (melanger) [inline .releaserc.json:5]: IMHO you can keep alpha and beta

File: package.json [hunk 5]
```diff
@@ -78,7 +78,17 @@
     "tinycolor2": "^1.4.2",
     "tslib": "^2.3.1",
     "y18n": "^5.0.8",
-    "zone.js": "0.11.4"
+    "zone.js": "0.11.4",
+    "@commitlint/cli": "^17.6.6",
```
Comment (melanger) [inline package.json:82]: Put these dependencies into devDependencies (`npm install --save-dev ...`)

File: .github/workflows/semantic-release.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+name: Semantic release
```
Comment (melanger) [inline .github/workflows/semantic-release.yml:1]: Do not create this file, release.yml already exists

File: package.json [hunk 7]
```diff
@@ -78,7 +78,12 @@
     "tinycolor2": "^1.4.2",
     "tslib": "^2.3.1",
     "y18n": "^5.0.8",
-    "zone.js": "0.11.4"
+    "zone.js": "0.11.4",
+    "@semantic-release/github": "^9.0.3",
+    "commitizen": "^4.3.0",
+    "conventional-changelog-conventionalcommits": "^6.1.0",
+    "inquirer": "^8.2.5",
+    "semantic-release": "^21.0.7"
```
Comment (melanger) [inline package.json:86]:
```suggestion
    "zone.js": "0.11.4"
```

remove this and run `npm update` to refresh package-lock.json

**Review by melanger — CHANGES_REQUESTED** [review]

**Review by melanger — APPROVED** [review]

**Comment by mattjoke** [comment]
Please change the node version to `lts/*` in release.yml, so it does not fail with the new versions of the semantic realease.

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**